### PR TITLE
feat(evaluator): implement relative offset resolution (#38)

### LINKED ISSUE #38: Evaluator: implement relative offset resolution
## Summary

Implement evaluation of relative offsets. The AST representation (`OffsetSpec::Relative`) already exists, but the evaluator returns a "not yet implemented" error at `src/evaluator/offset.rs:167`.

## Context

Relative offsets use the syntax `&+N` or `&-N` to reference positions relative to the end of the previous match. This enables sequential field parsing where each field follows the previous one without hardcoding absolute positions.

**Current state**: AST variant defined, evaluation stubbed with TODO.

## Acceptance Criteria

- [ ] `resolve_offset()` handles `OffsetSpec::Relative` using the previous match's end position
- [ ] Evaluation context tracks the current match position for relative offset resolution
- [ ] Positive and negative relative offsets work correctly
- [ ] Bounds checking prevents out-of-buffer access
- [ ] Unit tests for relative offset resolution
- [ ] Integration tests with nested rules using relative offsets

## Impact

**MEDIUM-HIGH** - Used in nested rule structures where child rules reference positions relative to parent matches.

## Files to Modify

- `src/evaluator/offset.rs` - Implement relative resolution (line ~167)
- `src/evaluator/mod.rs` - Track match position in evaluation context

## References

- GNU file magic(5) man page
- Epic #24

### PR DESCRIPTION
**Pull request by unclesp1d3r**
## Summary

Implements `OffsetSpec::Relative` evaluator support per #38 using GNU `file`/libmagic semantics. Threads a "previous match end" anchor through `EvaluationContext`, advances it after each successful match by the bytes consumed (variable-width types include c-string NUL terminators and pstring length prefixes), and resolves `Relative(delta)` as `last_match_end + delta` with bounds and overflow checks.

**Impact:** 21 files changed (1984 insertions, 213 deletions). 945 lib tests + 14 relative-offset integration tests + 11 other integration suites + 163 doctests, all green. Clippy clean with `-D warnings`. CI: 22 pass, 0 failures, mergeable.

**Review rounds completed:** 4 internal passes (ce:review autofix, pr-review-toolkit ×2, Copilot PR reviewer). 0 critical/high findings outstanding.

## Key design decisions

- **Anchor is global-shared across child recursion, not save/restored.** Siblings see the anchor wherever the deepest descendant of the previous sibling left it — matches GNU `file` semantics. The anchor *value* may increase or decrease as successive rules match at different positions; it is not a high-watermark. Documented in `GOTCHAS.md` §3.8.
- **Top-level relative offsets resolve from anchor 0.** Non-negative deltas behave like `Absolute(N)`; negative deltas underflow to `InvalidOffset` (NOT interpreted like `Absolute(-N)`'s from-end semantics). Called out explicitly in the rustdoc for both `resolve_offset` and `evaluate_single_rule`.
- **Pascal-string consumption is clamped against the remaining buffer.** Prevents attacker-controlled length prefixes (e.g., `\xFF\xFF\xFF\xFF`) from poisoning the anchor to `usize::MAX` and silently suppressing all subsequent relative-offset rules. Found by adversarial review (SEC-001 / ADV-001), regression-tested with both BE and LE oversized-prefix cases. Documented as a reusable learning in `docs/solutions/security-issues/pstring-anchor-poisoning.md`.
- **Fixed-width `bytes_consumed` is also bounds-checked.** Per Copilot reviewer feedback, added `offset.checked_add(width) <= buffer.len()` guard so the helper is infallible regardless of caller discipline — no relying on "the engine only calls this after a successful read" as the only line of defense.
- **Engine-internal items are `pub(crate)`:** `bytes_consumed`, `last_match_end()` getter, `set_last_match_end()` setter, and `resolve_offset_with_context` are deliberately not stable API surface. The anchor-threading contract is internal and shouldn't lock in semver guarantees.

## Behavior changes

`OffsetSpec::Relative` previously returned `EvaluationError::UnsupportedType`. It now resolves:

| Entry point | `Relative(N)` for N ≥ 0 | `Relative(-N)` for N > 0 |
|---|---|---|
| `evaluate_rules` + `EvaluationContext` | Resolves against actual `last_match_end + N` | `last_match_end - N` if in bounds, else `InvalidOffset` |
| `evaluate_single_rule` (no context) | Resolves to absolute `N` | `InvalidOffset` (anchor 0 underflow) |
| `resolve_offset` (no context) | Same as `evaluate_single_rule` | Same as `evaluate_single_rule` |

Callers with existing error-handling code that pattern-matched `UnsupportedType` for relative offsets must remove that arm. Documented in the rustdoc for all three entry points.

## Implementation

- `EvaluationContext::last_match_end` field with `pub(crate)` getter/setter, cleared by `reset()` alongside `current_offset` and `recursion_depth`
- `evaluator::types::bytes_consumed` (`pub(crate)`) with buffer clamping for both fixed-width and variable-width (c-string, pstring) branches; `string_bytes_consumed` and `pstring_bytes_consumed` re-derive consumption directly from the buffer (not from `Value::String`, which can drift via `from_utf8_lossy`)
- `evaluator::offset::relative::resolve_relative_offset` with `isize::try_from` + `checked_add_signed` bounds, mapping over/underflow to `InvalidOffset` and out-of-buffer to `BufferOverrun`
- `evaluator::offset::resolve_offset_with_context` (`pub(crate)`) dispatcher; existing `resolve_offset` (`pub`) delegates with `last_match_end = 0` so existing call sites are unchanged
- `evaluator::engine::evaluate_single_rule_with_anchor` private helper; `evaluate_rules` updates the anchor before recursing into children, never saves/restores (GNU `file` global semantics)
- `evaluate_rules` graceful-skip arm for child-evaluation errors: `debug!` → `warn!` with a tightened comment noting the asymmetry (parent emitted, child silently dropped if the defensive arm fires)

## Test coverage

**Unit tests (30+ new):**
- 4 tests for `EvaluationContext::last_match_end` (field / clone / reset / set-get)
- 18 tests for `bytes_consumed` (fixed-width all types, c-string with/without NUL, max_length truncation, pstring 1/2/4-byte BE/LE, `/J` flag well-formed, `/J` underflow with multi-byte width, oversized prefix clamping BE+LE, boundary cases, `usize::MAX` overflow)
- 10 tests for `resolve_relative_offset` (positive/negative/zero deltas, top-level anchor=0, last-valid-index, underflow, overflow at `usize::MAX`, target past end, target == `buffer.len()`)
- 5 dispatcher tests in `offset/mod.rs` (relative via context, top-level default, passthrough for Absolute/FromEnd/Indirect)
- 3 tests in `engine/tests.rs` pinning `evaluate_single_rule` contracts (positive delta → absolute N, zero delta → start, negative delta → `Err(InvalidOffset)`)
- 1 test injecting a near-saturation anchor (`usize::MAX`) via the `pub(crate)` setter to verify all three Relative variants skip gracefully without panic

**Integration tests (14 in `tests/relative_offset_evaluation.rs`):**
- Numeric chain (parent + relative child)
- Positive/negative delta children
- Three-level chain marching forward
- String parent + relative child (NUL terminator accounting)
- PString parent + relative child (prefix accounting)
- Top-level zero anchor
- Sibling propagation at top level
- Non-matching intermediate sibling preserves anchor
- Anchor decreases when later sibling matches at lower position (non-monotonic semantics pinned)
- Context reset between buffers
- Out-of-bounds skip
- Underflow skip

## Documentation

- `AGENTS.md` — "Currently Implemented" / Module Organization / Quick Reference sections updated; `docs/solutions/` pointer added for knowledge-store discoverability
- `GOTCHAS.md` — new §3.8 documents the no-save-restore anchor semantics; §2.1 checklist updated to include `bytes_consumed` exhaustive-match requirement for new `TypeKind` variants
- `docs/src/evaluator.md` — Offset Resolution section rewritten for the two-case Relative behavior; `EvaluationContext` "Key Methods" split into Public vs Internal (`pub(crate)`) sections; Implementation Status checklist updated for #37 and #38
- `docs/solutions/security-issues/pstring-anchor-poisoning.md` — NEW bug-track solution doc capturing the SEC-001 finding, failed first attempts, generalizable rule ("when advancing internal state by attacker-controlled byte counts, clamp against actual buffer reality, not raw input"), and prevention strategies
- `docs/solutions/developer-experience/rust-test-visibility-boundary.md` — NEW knowledge-track doc capturing the lesson that tests needing `pub(crate)` items must live in `src/.../tests.rs`, not `tests/`

## Scope notes

A handful of files in the diff are incidental and not part of the #38 implementation:

- `.bun-version` + `mise.lock` + `mise.toml` — tool version pins that came in via merge-from-main
- `tessl.json` — unrelated `neonwatty/logo-designer-skill` dependency added in commit `b1f62fa`
- `docs/src/architecture.md` + `docs/src/compatibility.md` — Dosu auto-updates from commit `3cf5d5a`

Reviewers can skim these; they don't need deep review as part of the #38 work.

## Post-Deploy Monitoring & Validation

No additional operational monitoring required. This is a library change to evaluator semantics:

- **Runtime impact:** magic rules with `OffsetSpec::Relative` now evaluate correctly instead of erroring with `UnsupportedType`
- **API impact:** the public `resolve_offset` and `evaluate_single_rule` signatures are unchanged; internal items are `pub(crate)` and don't lock semver
- **Dependency impact:** no new Rust crates; the unrelated `tessl.json` change is documented above
- **Performance:** `bytes_consumed` adds one bounds check per successful match (negligible); `memchr` in `string_bytes_consumed` is O(n) over a region the existing `read_string` already scans, so worst case is a bounded 2x constant factor on string matches

Behavior change is documented in the rustdoc and `AGENTS.md` so downstream callers see the migration note inline.

## Review history

1. **Initial implementation** (`c9a89f1`) — feature + tests per the plan
2. **ce:review autofix pass** (`b7e5938`) — 12 safe_auto fixes including the SEC-001 pstring anchor poisoning fix, visibility narrowing to `pub(crate)`, doc gaps on context reuse
3. **Third-pass polish** (`4ebd9b7`) — pr-review-toolkit feedback: "monotonic" wording fix, `recursion_depth` in context-reuse warning, 2 new tests
4. **Copilot PR review** (`cd3a5e0`) — bounds guard on fixed-width `bytes_consumed`, Relative negative-delta clarification, `debug_assert!` wording fix, `docs/src/evaluator.md` Public/Internal method split
5. **Compound docs** (`41c75c3`) — pstring-anchor-poisoning learning + `docs/solutions/` discoverability pointer in AGENTS.md
6. **Fourth-pass pr-review-toolkit** (`58210ab`) — 6 findings: stale `resolve_offset` equivalence claim in `docs/src/evaluator.md`, `log::warn!` style consistency, `evaluate_single_rule(Relative(-1))` test coverage, stale comments, implementation status checklist
7. **Test visibility boundary learning** (`03f389b`) — Rust developer-experience solution doc capturing the `tests/` vs `src/.../tests.rs` gotcha hit during implementation

## Test Plan

- [x] All plan units complete (6 of 6)
- [x] All plan requirements (R1-R9) verified met
- [x] `just ci-check` passes locally
- [x] `cargo test` passes (945 lib + 14 integration + 11 other suites + 163 doctest)
- [x] `cargo clippy --all-targets -- -D warnings` clean
- [x] `cargo fmt --check` clean
- [x] 4 internal review rounds addressed
- [x] CI green on push (22 pass, 7 release skipping, 0 failures)
- [x] Maintainer review

Closes #38

### CODE REVIEW

File: src/evaluator/engine/mod.rs [hunk 1]
```diff
@@ -62,16 +62,48 @@ use log::debug;
...
+/// # Relative offset behavior
+///
+/// If the rule uses [`OffsetSpec::Relative`](crate::parser::ast::OffsetSpec::Relative),
+/// this function resolves it against anchor 0 -- equivalent to an absolute
+/// offset of the same delta. There is no "previous match" context when
+/// evaluating a single rule in isolation. Callers needing the GNU `file`
+/// anchor semantics (where relative offsets resolve against the end of the
+/// most recent match) should use [`evaluate_rules`] with an
+/// [`EvaluationContext`](crate::evaluator::EvaluationContext), which threads
+/// the anchor across the rule list.
```
[the remark this answers is not in the document]
Comment (unclesp1d3r) [reply src/evaluator/engine/mod.rs:83]: > The `evaluate_single_rule` rustdoc says `OffsetSpec::Relative` resolves against anchor 0 and is "equivalent to an absolute offset of the same delta". That equivalence breaks for negative deltas...

Addressed in `cd3a5e0`. Reworded the rustdoc to spell out the two cases explicitly: non-negative delta resolves like `Absolute(N)` from start of buffer, while negative delta underflows the anchor and returns `EvaluationError::InvalidOffset` (which is *not* the same as `Absolute(-N)` from-end semantics). Same fix applied to `resolve_offset` in `src/evaluator/offset/mod.rs` which had the identical misleading wording.

File: src/evaluator/types/mod.rs [hunk 2]
```diff
@@ -171,5 +171,156 @@ pub fn coerce_value_to_type(value: &Value, type_kind: &TypeKind) -> Value {
...
+///   or 4 bytes, BE/LE), accounts for the `/J` flag (stored length includes
+///   prefix width), caps by `max_length`, and returns `prefix_width +
+///   actual_payload_bytes`. The result is also clamped against the remaining
+///   buffer length so a malicious oversized length prefix cannot poison the
+///   anchor.
+#[must_use]
+pub(crate) fn bytes_consumed(buffer: &[u8], offset: usize, type_kind: &TypeKind) -> usize {
+    if let Some(bits) = type_kind.bit_width() {
+        return (bits as usize) / 8;
+    }
```
[the remark this answers is not in the document]
Comment (unclesp1d3r) [reply src/evaluator/types/mod.rs]: > Consider adding a guard like `if offset >= buffer.len() { return 0; }` before the `bit_width()` fast path (and updating the tests accordingly) so the function matches its documented defensive behavior and can't advance the anchor when misused.

Addressed in `cd3a5e0`. Took your suggestion and added the guard rather than papering over the gap with documentation. The fixed-width branch now does `offset.checked_add(width)` and returns `0` if the result exceeds `buffer.len()`, mirroring the variable-width path. The function is now genuinely infallible regardless of caller discipline. Test `test_bytes_consumed_fixed_width_returns_zero_past_end` was rewritten to pin the new bounds-checked behavior at the boundary (offset==buf.len(), way past end, last valid index, multi-byte type at boundary, and `usize::MAX` overflow).

File: docs/src/evaluator.md [hunk 3]
```diff
@@ -63,10 +65,12 @@ Note: Fields are private; use accessor methods like `current_offset()`, `recursi
 
 - `new()` - Create context with default configuration
 - `current_offset()` / `set_current_offset()` - Track current buffer position
+- `last_match_end()` - Get end offset of most recent match (used for relative offset resolution)
+- `set_last_match_end(pos: usize)` - Set the previous-match anchor (used internally by the engine)
 - `recursion_depth()` - Query current recursion depth
```
[the remark this answers is not in the document]
Comment (unclesp1d3r) [reply docs/src/evaluator.md:68]: > The docs list `last_match_end()` / `set_last_match_end(...)` as "Key Methods", but in code these are `pub(crate)` and therefore not callable by external users of the library. Please clarify in this section that these methods are internal-only (or remove them from the public-facing method list)...

Addressed in `cd3a5e0`. Split the method list into "Public Methods" and "Internal (`pub(crate)`) — engine use only" sections so external readers see at a glance which methods are callable. The internal section explicitly directs users to `evaluate_rules` (or `evaluate_rules_with_config`) with an `EvaluationContext` for relative-offset behavior, and notes that `EvaluationContext::reset()` clears the anchor alongside the other mutable state.

File: src/evaluator/types/mod.rs [hunk 4]
```diff
@@ -171,5 +171,176 @@ pub fn coerce_value_to_type(value: &Value, type_kind: &TypeKind) -> Value {
...
+            *max_length,
+            *length_width,
+            *length_includes_itself,
+        ),
+        // A new variable-width TypeKind variant was added without updating
+        // this match. Returning 0 here would silently corrupt the GNU `file`
+        // anchor for any rule using relative offsets after a match of the
+        // new type. The debug assertion fires immediately in test/dev
+        // builds; release builds keep the 0 fallback (graceful skip rather
+        // than panic), but the asserting log highlights the gap.
```
[the remark this answers is not in the document]
Comment (unclesp1d3r) [reply src/evaluator/types/mod.rs]: > The comment says "the asserting log highlights the gap", but `debug_assert!` doesn't log and is compiled out in release builds; it panics in debug builds and does nothing in release. Please adjust the wording to reflect the actual behavior...

Addressed in `cd3a5e0`. Reworded the comment to reflect the actual behavior: the `debug_assert!` panics in test/dev builds so the gap is caught loudly during testing, while release builds compile it out and keep the 0 fallback (graceful skip rather than panic), so the case is only surfaced by the assertion in non-release builds. Thanks for the precise rewrite — applied essentially as suggested.

File: src/evaluator/types/mod.rs [hunk 5]
```diff
@@ -171,5 +171,193 @@ pub fn coerce_value_to_type(value: &Value, type_kind: &TypeKind) -> Value {
...
 }
 
+/// Returns the number of buffer bytes a successful `read_typed_value` would
+/// consume for the given `type_kind` at `offset`.
+///
+/// This mirrors the consumption logic of the underlying read functions and is
+/// used by the evaluation engine to advance the GNU `file` "previous match"
+/// anchor for relative offset resolution. It is `pub(crate)` because no
+/// external caller should depend on the anchor-advance contract -- the only
+/// intended caller is `evaluate_rules` in the engine.
```
[the remark this answers is not in the document]
Comment (unclesp1d3r) [reply src/evaluator/types/mod.rs]: > the string case intentionally *adds* 1 byte for the NUL terminator. `read_string` (in `src/evaluator/types/string.rs`) stops at the NUL and does not include it in the bytes read/returned, so this wording is misleading. Please reword to describe this as "anchor advance distance"...

Addressed in `b70ec40`. Rewrote the `bytes_consumed` rustdoc header to call the return value "anchor-advance distance" and explicitly note that it may include framing bytes (NUL terminators, pstring length prefixes) that the underlying read helpers do not return. Added a sentence telling callers not to equate the value with "bytes `read_typed_value` returned" since it's a superset for variable-width types.

File: src/evaluator/types/mod.rs [hunk 6]
```diff
@@ -171,5 +171,193 @@ pub fn coerce_value_to_type(value: &Value, type_kind: &TypeKind) -> Value {
...
+}
+
+/// Compute the buffer bytes consumed by a successful c-string read.
+///
+/// Mirrors `read_string`: scans from `offset` for the first NUL within
+/// `max_length` bytes (or to the end of the buffer when `max_length` is
+/// `None`), and returns `length_to_nul + 1` when a NUL was found, or
+/// `length_read` when no NUL was found (truncated by buffer end or
+/// `max_length`).
+///
```
[the remark this answers is not in the document]
Comment (unclesp1d3r) [reply src/evaluator/types/mod.rs]: > `string_bytes_consumed`'s doc comment says it "Mirrors `read_string`", but it returns `length_to_nul + 1` when a NUL is found, whereas `read_string` reads up to (but not including) the NUL. Consider adjusting the wording to avoid implying these two functions count consumed bytes the same way...

Addressed in `b70ec40`. Reworded the `string_bytes_consumed` doc comment. It now says the function "uses the same scan logic as `read_string`" but explicitly notes that unlike the `Value::String` returned by `read_string` (which excludes the NUL terminator), this helper counts the NUL terminator as consumed for anchor-advancement purposes. Added a "do not fix this to align with `read_string`'s byte count -- the asymmetry is the point" warning so a future maintainer doesn't regress it.

[the remark this answers is not in the document]
Comment (unclesp1d3r) [reply src/evaluator/types/mod.rs]: *(this is a duplicate of the fifth-round `string_bytes_consumed` comment at line 274; replying here for consistency)*

File: src/evaluator/types/mod.rs [hunk 7]
```diff
@@ -171,5 +171,193 @@ pub fn coerce_value_to_type(value: &Value, type_kind: &TypeKind) -> Value {
...
+/// The `+ 1` for the NUL-found case is intentional: a `Relative(0)` rule
+/// following a NUL-terminated string match resolves to the byte
+/// *immediately after* the NUL terminator, not the NUL itself. This
+/// matches GNU `file` semantics for chained record parsing.
```
[the remark this answers is not in the document]
Comment (unclesp1d3r) [reply src/evaluator/types/mod.rs]: > `string_bytes_consumed` claims it "Mirrors `read_string`" and returns `length_to_nul + 1` when a NUL is found, but `read_string` currently returns a `Value::String` whose length is `length_to_nul` (the terminator is not included)... please adjust this doc comment to clarify it uses the same scan logic as `read_string` but counts the NUL terminator as consumed for relative-offset anchoring

Addressed in `b70ec40`. Applied essentially the suggested rewrite: the doc now says the helper "uses the same scan logic as `read_string`" and explicitly contrasts with the `Value::String` that `read_string` returns (which excludes the terminator). Also added the "Counting the terminator is intentional for relative-offset anchoring" paragraph with a "do not fix this to align" note. Matches GNU `file` semantics for chained record parsing.

File: docs/src/architecture.md [hunk 8]
```diff
@@ -165,7 +165,7 @@ The evaluator executes magic rules against file buffers to identify file types.
   - `mod.rs`: Dispatcher (`resolve_offset`) and re-exports
   - `absolute.rs`: `OffsetError`, `resolve_absolute_offset`
   - `indirect.rs`: `resolve_indirect_offset` stub (issue #37)
```
[the remark this answers is not in the document]
Comment (unclesp1d3r) [reply docs/src/architecture.md]: > The evaluator module structure list still says `offset/indirect.rs` is a stub (issue #37), but the file currently contains a full `resolve_indirect_offset` implementation.

Addressed in `b70ec40`. Updated the entry to `indirect.rs`: `resolve_indirect_offset` for indirect pointer-based offset resolution (issue #37, shipped). This is the same stale-stub pattern I fixed in `docs/src/evaluator.md` in a prior review round, just in a different file; good catch that architecture.md needed the same treatment.

File: GOTCHAS.md [hunk 9]
```diff
@@ -71,6 +71,12 @@ The nom `tuple` combinator is deprecated. Use bare tuple syntax `(a, b, c)` dire
...
+
+The load-bearing invariant is that the anchor is updated *before recursing into children* (so children and their followers see the new anchor). The current code also happens to set the anchor before `matches.push(...)`, but the push-ordering relative to `set_last_match_end` is incidental for anchor correctness -- only the ordering before the `evaluate_rules` recursion call matters. (Future code that reads the anchor while iterating `matches` would make this ordering load-bearing, so do not "optimize" the order without checking call sites first.) `bytes_consumed()` (in `evaluator/types/mod.rs`) is the source of truth for advance distance; for variable-width types it re-derives consumption from the buffer rather than trusting `Value::String.len()` (which can drift from the original byte length via `from_utf8_lossy`). Pascal-string consumption is also clamped against the remaining buffer to prevent attacker-controlled length prefixes from poisoning the anchor to `usize::MAX`.
```
[the remark this answers is not in the document]
Comment (unclesp1d3r) [reply GOTCHAS.md]: > Section title "Relative Offsets: Anchor is Global-Monotonic" is mathematically inaccurate given the documented/implemented behavior that the anchor can move backward when a later match ends earlier. Consider renaming this to something like "Global/shared anchor (no save/restore)"

Addressed in `b70ec40`. Renamed S3.8 to "Relative Offsets: Global Anchor, No Save/Restore" and reworked the body text to make clear that the anchor is global/shared (not stack-scoped) but its numeric value is not guaranteed non-decreasing — a `Relative(-N)` match or any later rule at a lower absolute position can move it earlier. The "monotonic" wording is now gone.

File: GOTCHAS.md [hunk 10]
```diff
@@ -71,6 +71,12 @@ The nom `tuple` combinator is deprecated. Use bare tuple syntax `(a, b, c)` dire
 
 Lowercase pointer specifiers (`.s`, `.l`, `.q`) map to **little-endian**, not native endian. Uppercase (`.S`, `.L`, `.Q`) map to big-endian. All numeric pointer types are **signed by default** (per S6.3). The adjustment is parsed **after** the closing paren: `(base.type)+adj`, not `(base.type+adj)`.
 
+### 3.8 Relative Offsets: Anchor is Global-Monotonic, No Save/Restore
```
[the remark this answers is not in the document]
Comment (unclesp1d3r) [reply GOTCHAS.md]: > The GOTCHAS section title says the relative-offset anchor is "Global-Monotonic", but elsewhere (including `EvaluationContext` docs) the anchor is explicitly allowed to increase or decrease as later rules match at lower/higher positions. Consider renaming this header

Addressed in `b70ec40`. Applied essentially the suggested rename (used "Global Anchor, No Save/Restore" which matches the `EvaluationContext` field doc phrasing). Also reworded the body to explicitly state the non-monotonic property.
